KlimaDAO/klimadao
# Makka/deposit carbon view

## Description

New view added to the dApp which allows users to deposit TCO2 tokens in exchange for BCT tokens 1-1.

## Related Ticket

Closes #2368 

## How to Test

1. Navigate to the deposit view. Connect a wallet which contains some TCO2 tokens which you wish to exchange for BCT.
2. The first valid token in the list of user tokens should be shown on the UI and the amount available next to it. 
3. Click on the token in the UI and you should see a popup/modal with all available tokens visible. If a token is not eligible i.e. is a C3T token, you should still see it in the popup but should not be able to select it.
4. Enter the amount you wish to exchange, if the user enters more than what is available you shouldn't be able to continue and the text should change to the color red like the figma designs.
5. If the amount entered is valid, you should be able to proceed to the approval step. Approve the token and submit (similar to the other flows across the dapp).
